_inc/lib/admin-pages/class.jetpack-my-jetpack-page.php

```diff
@@ -46,11 +46,34 @@ function jetpack_my_jetpack_change_user() {
 				Jetpack::init()->stat( 'admin', 'change-primary-successful' );
 				Jetpack::init()->do_stats( 'server_side' );
 
+				// Change the blog owner dotcom side
+				$this->tell_dotcom_about_user_switch( $new_master_user );
 			}
 		}
 	}
 
 	/*
+	 * Tell wpcom that the master user has switched
+	 * so we can update the 'wpcom_blog_owner'
+	 */
+	function tell_dotcom_about_user_switch( $new_master ) {
+		$id = Jetpack::get_option( 'id', false );
```

Comment:
The fact that we're passing in the Blog ID here feels hacky and bad.  Let's just have the wpcom side tell the blog ID based on the authentication tokens.

You can get the ID wpcom side like this:

```php
global $wp_xmlrpc_server;
$blog_id = (int) $wp_xmlrpc_server->token->blog_id;
```